test/driver-e2e-specs.js

```diff
@@ -75,6 +76,43 @@ describe('FakeDriver - via HTTP', () => {
       await B.delay(250);
       await driver.source().should.eventually.be.rejectedWith(/terminated/);
     });
+
+    it('should accept W3C capabilities', async () => {
+      // Try with valid capabilities and check that it returns a session ID
+      const w3cCaps = {
+        capabilities: {
+          alwaysMatch: {platformName: 'Fake'},
+          firstMatch: [{'appium:deviceName': 'Fake', 'appium:app': TEST_FAKE_APP}],
+        }
+      };
+
+      const { status, value, sessionId } = await request.post({url: `http://${TEST_HOST}:${TEST_PORT}/wd/hub/session`, form: w3cCaps, json: true});
+      status.should.equal(0);
+      sessionId.should.be.a.string;
+      value.should.exist;
+
+      // Now use that sessionId to call /screenshot
+      const { status:screenshotStatus, value:screenshotValue } = await request({url: `http://${TEST_HOST}:${TEST_PORT}/wd/hub/session/${sessionId}/screenshot`, json: true});
+      screenshotValue.should.equal('hahahanotreallyascreenshot');
+      screenshotStatus.should.equal(0);
+
+      // Now use that sessionID to call an arbitrary W3C-only endpoint that isn't implemented to see if it throws correct error
+      await request.post({url: `http://${TEST_HOST}:${TEST_PORT}/wd/hub/session/${sessionId}/execute/async`, form: {script: '', args: ['a']}, json: true}).should.eventually.be.rejectedWith(/not yet been implemented/);
```

Comment:
not sure but i think you can just set the request body via the `json` param, rather than setting it via `form` and then also setting `json` to `true`?